codersdk/workspaceagents.go

```diff
@@ -492,6 +492,11 @@ func (c *Client) WatchWorkspaceAgentMetadata(ctx context.Context, id uuid.UUID)
 				firstEvent = false
 			}
 
+			// Ignore pings.
+			if sse.Type == ServerSentEventTypePing {
+				continue
+			}
```

Comment:
This was a bug in `WatchWorkspaceAgentMetadata`, leading to hitting an error below (nil data) and breaking the watcher. This could be a/the cause that the agent metadata tests have been flakey in the past.